rmw_ecal_shared_cpp/src/internal/publisher.hpp

```diff
@@ -55,14 +55,15 @@ namespace eCAL
                                       type_support_->GetTypeDescriptor());
         publisher_.SetQOS(qos.ecal_qos);
         publisher_.SetAttribute("node_name", node_name);
-	publisher_.SetAttribute("node_namespace", node_namespace);
-	publisher_.AddEventCallback(eCAL_Publisher_Event::pub_event_dropped, std::bind(&Publisher::OnDataDropped, this, _1, _2));
+	      publisher_.SetAttribute("node_namespace", node_namespace);
+	      publisher_.AddEventCallback(eCAL_Publisher_Event::pub_event_dropped, std::bind(&Publisher::OnDataDropped, this, _1, _2));
       }
 
       void Publish(const void *data)
       {
         auto serialized_data = type_support_->Serialize(data);
-        publisher_.Send(serialized_data.data(), serialized_data.size());
+        auto publish_timestamp = eCAL::Time::GetNanoSeconds();
+        publisher_.Send(serialized_data.data(), serialized_data.size(), publish_timestamp);
```

Comment:
It appears you modified this call to `Send` but did not modify the function signature or definition.

Reply:
This is the Send func in usr/include/ecal/ecal_publisher.h:
```
    /**
     * @brief Send a message to all subscribers. 
     *
     * @param buf_    Pointer to content buffer. 
     * @param len_    Length of buffer. 
     * @param time_   Send time (-1 = use eCAL system time in us, default = -1).
     *
     * @return  Number of bytes sent. 
    **/
    size_t Send(const void* const buf_, size_t len_, long long time_ = -1) const;
```
It is using microsecond as unit in default. Which does not mach timestamp of recieved timestamp (nanosec).

Reply:
Ah, I got ya. I didn't realize it was a defaulted argument. Carry on.

Reply:
Actually, this still needs to be changed because the function definition specifies microsceonds. Please at least use `GetMicroSeconds`.

Reply:
Wait. This shouldn't be necessary at all. If you just leave it defaulted, it will get the system time in the correct units. Just remove line 65.